app/models/service_ansible_playbook.rb

```diff
@@ -83,14 +83,22 @@ def get_job_options(action)
   end
 
   def save_job_options(action, overrides)
-    job_options = options.fetch_path(:config_info, action.downcase.to_sym).slice(:hosts, :extra_vars).with_indifferent_access
+    option_keys = %i[
+      hosts
+      extra_vars
+      credential_id
+      vault_credential_id
+      network_credential_id
+      cloud_credential_id
+    ]
+    job_options = options.fetch_path(:config_info, action.downcase.to_sym).slice(*option_keys).with_indifferent_access
     job_options[:extra_vars].try(:transform_values!) do |val|
       val.kind_of?(String) ? val : val[:default] # TODO: support Hash only
     end
     job_options.deep_merge!(parse_dialog_options) unless action == ResourceAction::RETIREMENT
     job_options.deep_merge!(overrides)
 
-    %i[credential vault_credential].each do |cred|
+    %i[credential vault_credential network_credential cloud_credential].each do |cred|
```

Comment:
Also, should this list be shared with https://github.com/ManageIQ/manageiq/pull/18969/files#diff-96e20f5faa69bc61958955255cb5341fR54 ?

Reply:
I wish, but I don't think we can do that. In the long term I think the configuration script class will probably just go away.